src/main/java/org/springframework/data/r2dbc/repository/query/LikeEscaper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.r2dbc.repository.query;
+
+import org.springframework.lang.Nullable;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Helper class encapsulating an escape character for LIKE queries and the actually usage of it in escaping
+ * {@link String}s.
+ * <p>
+ * This class is an adapted version of {@code org.springframework.data.jpa.repository.query.EscapeCharacter} from
+ * Spring Data JPA project.
+ *
+ * @author Roman Chigvintsev
+ */
+public class LikeEscaper {
```

Comment:
This type looks as it would be a good candidate for Spring Data Relational, to be defined as part of a `Dialect`. Please note that SQL Server (probably the only exception for now) uses square brackets (`[]`) for a regex-y style declaring character ranges. We should design the class with this behavior in mind. Turning this class into an interface and adding a default implementation for `%` would be a good first step. In the SQL Server-specific dialect, we can add an adaption for square brackets.